frontend-react/src/components/Admin/AdminReceiverDashboard.tsx

```diff
@@ -401,31 +425,50 @@ function MainRender(props: {
                 const sliceClassName =
                     SUCCESS_RATE_CLASSNAME_MAP[successForSlice.currentState];
 
-                // we can use the currentkey in the data dict as a unique key for this cell
+                const sliceFilterClassName =
+                    MATCHING_FILTER_CLASSNAME_MAP[filteredSlice];
+
+                // it's possible to match more than on data entry per slice
+                const dataCount =
+                    successForSlice.countSuccess + successForSlice.countFailed;
+                const dataOffset = offset - dataCount;
+                const dataOffsetEnd = offset;
+
                 sliceElements.push(
                     <Grid
                         row
                         key={`slice:${currentReceiver}|${timeSlotStart}`}
-                        className={`slice ${sliceClassName}`}
-                        data-keyoffset={keyOffset - 1}
+                        className={`slice ${sliceClassName} ${sliceFilterClassName}`}
+                        data-offset={dataOffset}
+                        data-offset-end={dataOffsetEnd}
                         onClick={
                             successForSlice.currentState ===
                             SuccessRate.UNDEFINED
                                 ? undefined // do not even install a click handler noop
                                 : (evt) => {
-                                      // get saved offset from "data-keyoffset" attribute on this element
-                                      const dataKeyOffset = parseInt(
-                                          evt.currentTarget?.dataset[
-                                              "keyoffset"
-                                          ] || "-1"
+                                      // get saved offset from "data-offset" attribute on this element
+                                      const target = evt.currentTarget;
+                                      const sliceStart = parseInt(
+                                          target?.dataset["offset"] || "-1"
```

Comment:
any reason not to do?
```suggestion
                                          target?.dataset.offset || "-1"
```

Reply:
Just consistency with the other dataset variables all being quoted since they have "-" names. If I make these props on a Slice component, these go away.